### PR TITLE
feat: [CI-3878]: add always run annotation

### PR DESCRIPTION
**Pull request by jamie-harness**
You can use the following comments to re-trigger PR Checks

- Compile: `trigger compile`
- runAeriformCheck: `trigger AeriformCheck`
- CodeFormat: `trigger codeformat`
- MessageMetadata: `trigger messagecheck`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- runDockerizationCheck: `trigger dockerizationcheck`
- runAuthorCheck: `trigger authorcheck`
- Checkstyle: `trigger checkstyle`
- PMD: `trigger pmd`
- TI-bootstrap: `trigger ti0`
- TI-bootstrap1: `trigger ti1`
- TI-bootstrap2: `trigger ti2`
- TI-bootstrap3: `trigger ti3`
- TI-bootstrap4: `trigger ti4`
- FunctionalTest1: `trigger ft1`
- FunctionalTest2: `trigger ft2`
- CodeBaseHash: `trigger codebasehash`

---
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/32721)

### CODE REVIEW

**Comment by jamie-harness** [comment]
trigger messagecheck

**Comment by jamie-harness** [comment]
trigger messagecheck

**Review by hemanthmantri — APPROVED** [review]
> Reviewed 2 of 2 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/32721)* status: :shipit: complete! ✓

**Review by rutvijmehta-harness — APPROVED** [review]
LG

**Review by rutvijmehta-harness — APPROVED** [review]
> Reviewed all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/32721)* status: :shipit: complete! ✓

File: product/ci/addon/parser/cg/callgraph.go [hunk 1]
```diff
@@ -96,6 +97,8 @@ func FromStringMap(data map[string]interface{}) (*Callgraph, error) {
 							node.Class = v.(string)
 						case "callsReflection":
 							node.CallsReflection = v.(bool)
+						case "alwaysRun":
+                            node.AlwaysRun = v.(bool)
```
Comment (hemanthmantri) [inline product/ci/addon/parser/cg/callgraph.go:101]: Indent.

File: product/ci/addon/parser/cg/callgraph_test.go [hunk 2]
```diff
@@ -89,6 +90,8 @@ func getCgObject(mp map[string]interface{}) ([]Node, []Relation, []Relation) {
 							node.Class = v.(string)
 						case "callsReflection":
 							node.CallsReflection = v.(bool)
+						case "alwaysRun":
+                            node.AlwaysRun = v.(bool)
```
Comment (hemanthmantri) [inline product/ci/addon/parser/cg/callgraph_test.go:94]: Indent

File: product/ci/addon/parser/cg/callgraph.go (hunk 1 above)
Comment (jamie-harness) [reply product/ci/addon/parser/cg/callgraph.go:101]: some of those indentions are caused by space vs indent. I am following harness-core standard here. Might need to change everything.

**Pull request merged**